### ISSUE TITLE
Thonny does not yet format "match" and "case"

### ISSUE THREAD

**Issue by TOAGGA** [thonny/thonny#2358]
Python 3.10 introduced structural pattern matching ([PEP 634](https://peps.python.org/pep-0634/)).

Python has already started formatting these statements in a nice bold colour…

![New to markdown.](https://i.ibb.co/64R2q4D/Screen-Shot-2022-07-21-at-21-05-47.png)
just like it formats if, import, else, etc., statements.
![](https://i.ibb.co/HpHshj3/Screen-Shot-2022-07-21-at-21-11-36.png)

Therefore, it makes sense that thonny should also format match and case this way.
For example, look here:

![](https://i.ibb.co/K6rvhZp/Screen-Shot-2022-07-21-at-21-15-02.png)

In other themes:
![](https://i.ibb.co/098FGmH/Screen-Shot-2022-07-21-at-21-17-36.png)
![](https://i.ibb.co/KNzhZHr/Screen-Shot-2022-07-21-at-21-19-12.png)

TLDR:
Thonny doesn’t format match and case correctly.

(Also, I’m sorry. I’m new to github and markdown so this is probably not the prettiest issue submission.)

**Comment by aivarannamaa** [comment]
Thank you for the report and sorry for my late response! 

I'll see how to solve this. The problem is that `match` and `case` are _soft keywords_ -- they can be also used as variable names, so the coloring procedure must take their context into account.
